src/include/duckdb.h

```diff
@@ -4485,6 +4522,99 @@ DUCKDB_C_API char *duckdb_table_description_get_column_name(duckdb_table_descrip
+                                                          duckdb_data_chunk chunk, duckdb_arrow_array *out_arrow_array);
+
+/*!
+Transforms an Arrow Schema into a DuckDB Schema.
+
+* @param connection The connection to get the transformation settings from.
+* @param schema The input Arrow schema. Must be destroyed with `duckdb_destroy_arrow_schema`.
+* @param out_types The Arrow converted schema with extra information about the Arrow Types. Must be destroyed with
+`duckdb_destroy_arrow_converted_schema`.
+* @param out_names The resulting column names array. Must be appropriately deleted.
+* @param out_column_count The number of columns extracted from the Arrow schema.
+* @return The error data.
+*/
+DUCKDB_C_API duckdb_error_data arrow_to_duckdb_schema(duckdb_connection connection, duckdb_arrow_schema schema,
+                                                      duckdb_arrow_converted_schema *out_types, char ***out_names,
+                                                      idx_t *out_column_count);
+
+/*!
+Transforms an Arrow array into a DuckDB data chunk. The data chunk will retain ownership of the underlying Arrow data.
+ However, arrow_array must still be destroyed with `duckdb_destroy_arrow_array`
+
+* @param connection The connection to get the transformation settings from.
+* @param arrow_array The input Arrow array. Even though data ownership is passed on to DuckDB's DataChunk, the
+underlying object must be destroyed with 'duckdb_arrow_array'.
+* @param converted_schema The Arrow converted schema with extra information about the Arrow Types.
+* @param out_chunk The resulting DuckDB data chunk. Must be destroyed by duckdb_destroy_data_chunk.
+* @return The error data.
+*/
+DUCKDB_C_API duckdb_error_data arrow_to_duckdb_data_chunk(duckdb_connection connection, duckdb_arrow_array arrow_array,
+                                                          duckdb_arrow_converted_schema converted_schema,
+                                                          duckdb_data_chunk *out_chunk);
+
+/*!
+Destroys the arrow schema and de-allocates all memory allocated for that schema.
+
+* @param arrow_schema The arrow schema to destroy.
+*/
+DUCKDB_C_API void duckdb_destroy_arrow_schema(duckdb_arrow_schema *arrow_schema);
+
+/*!
+Destroys the arrow converted schema and de-allocates all memory allocated for that arrow converted schema.
+
+* @param arrow_converted_schema The arrow converted schema to destroy.
+*/
+DUCKDB_C_API void duckdb_destroy_arrow_converted_schema(duckdb_arrow_converted_schema *arrow_converted_schema);
+
+/*!
+Destroys the arrow array and de-allocates all memory allocated for that arrow array.
+
+* @param arrow_array The arrow_array to destroy.
+*/
+DUCKDB_C_API void duckdb_destroy_arrow_array(duckdb_arrow_array *arrow_array);
+
+/*!
+Creates an empty  arrow array.
+The result must be destroyed with `duckdb_destroy_arrow_array`.
+
+* @return The arrow array.
+*/
+DUCKDB_C_API duckdb_arrow_array duckdb_create_arrow_array();
```

Comment:
This is not really useful, we'd want to be able to take in an existing `ArrowArray *` and take ownership of it
But to do that we'd need to put these methods behind the define of the spec: https://arrow.apache.org/docs/format/CDataInterface.html#structure-definitions

`ARROW_C_DATA_INTERFACE`